AdminWebsite/AdminWebsite/Controllers/HearingsController.cs

```diff
@@ -110,18 +110,15 @@ public async Task<ActionResult<HearingDetailsResponse>> Post([FromBody] BookNewH
                 string adUserId;
                 if(string.IsNullOrWhiteSpace(participant.Username))
                 {
-                    // create user
-                    var newUser = await _userAccountService.CreateNewUserInAD(participant);
-                    participant.Username = newUser.Username;
-                    adUserId = newUser.User_id;
+                    adUserId = await _userAccountService.UpdateParticipantUsername(participant);
                 }
                 else
                 {
                     // get user
                     adUserId = await _userAccountService.GetAdUserIdForUsername(participant.Username);
                 }
-                    
-                usernameAdIdDict[participant.Username] = adUserId;
+                // username's participant will be set by this point
+                usernameAdIdDict[participant.Username!] = adUserId;
```

Comment:
No need for Null forgiving here as a null string will throw when indexing into a dic

Reply:
The IDE kept raising warning here, I forgave it because the compiler could not recognise the value would be set by useraccount service